From 73e0a5a422f7c484a083c909e3b573429b7c3301 Mon Sep 17 00:00:00 2001
From: Thiago Macieira <thiago.macieira@intel.com>
Date: Thu, 19 Nov 2020 10:27:54 -0800
Subject: [PATCH 089/171] QCoreApplication: add more information to
 processEvents() docs

Just shows how bad an idea to call this function it is.

Fixes: QTBUG-88653
Change-Id: If51855da004b4f3fbf43fffd1648fc294065160d
Reviewed-by: Lars Knoll <lars.knoll@qt.io>
(cherry picked from commit 8ec9a48e80a72c7b9e82107a2c66de6cbf7a02fe)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/kernel/qcoreapplication.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/corelib/kernel/qcoreapplication.cpp b/src/corelib/kernel/qcoreapplication.cpp
index f761696aea..55316a79de 100644
--- a/src/corelib/kernel/qcoreapplication.cpp
+++ b/src/corelib/kernel/qcoreapplication.cpp
@@ -1300,6 +1300,9 @@ void QCoreApplication::processEvents(QEventLoop::ProcessEventsFlags flags)
     \note Unlike the \l{QCoreApplication::processEvents(QEventLoop::ProcessEventsFlags flags)}{processEvents()}
     overload, this function also processes events that are posted while the function runs.
 
+    \note All events that were queued before the timeout will be processed,
+    however long it takes.
+
     \threadsafe
 
     \sa exec(), QTimer, QEventLoop::processEvents()
-- 
2.31.1

